### PR TITLE
refactor: improve error messages

### PR DESCRIPTION
**Pull request by dawnseeker8**
Currently the `ledger-keyring` throws the error `unknown error` when user locks their ledger or didn't turn open the `eth` app in ledger in MM extension. This PR changes to provide a better error message to indicate what happen.

### CODE REVIEW

File: .gitignore [hunk 1]
```diff
@@ -76,3 +76,6 @@ node_modules/
 !.yarn/releases
 !.yarn/sdks
 !.yarn/versions
+
+# exclude webstorm profile file
+.idea/
```
Comment (gantunesr) [inline .gitignore:81]: Is this required?

File: src/ledger-keyring.ts [hunk 2]
```diff
@@ -212,7 +212,9 @@ export class LedgerKeyring extends EventEmitter {
         hdPath: path,
       });
     } catch (error) {
-      throw error instanceof Error ? error : new Error('Unknown error');
+      throw error instanceof Error
+        ? error
+        : new Error('Unlock your Ledger device and open the ETH app');
```
Comment (gantunesr) [inline src/ledger-keyring.ts:217]: How about this message?

```suggestion
        : new Error('Ledger ETH app is closed. Open it to unlock.');
```

File: .gitignore (hunk 1 above)
Comment (gantunesr) [reply .gitignore:81]: nvm

File: src/ledger-keyring.ts (hunk 2 above)
Comment (dawnseeker8) [reply src/ledger-keyring.ts:217]: The current error message was defined by Vivek, and for some reason, Extension will throw `Unknown Error` even ledger is locked. that is why we have Unlock your ledger first in error message, and then ask user to open ETH app.

Comment (gantunesr) [reply src/ledger-keyring.ts:217]: Is this error being used in the UI?

**Review by gantunesr — DISMISSED** [review]

File: src/ledger-keyring.ts (hunk 2 above)
Comment (dawnseeker8) [reply src/ledger-keyring.ts:217]: yes, this error will display in `select hardware screen` in metamask extension. i dont know why we didn't do the localisation translation for those error messsage in frontend.

Comment (angelcheung22) [reply src/ledger-keyring.ts:218]: @AlexJupiter any preference which we shall use?

Comment (gantunesr) [reply src/ledger-keyring.ts:218]: > I don't know why we didn't do the localisation translation for those error messsage in frontend.

This was my point, user facing errors should be added to the files for translation. This error is just for developers I recommend to use `Ledger Ethereum app is closed. Open it to unlock.` for development error purposes and catch this error in the UI level to show a user friendly message.

Overall I agree with @dawnseeker8 approach regarding this

Comment (dawnseeker8) [reply src/ledger-keyring.ts:218]: i have changed the message to `Ledger Ethereum app closed. Open it to unlock.`

**Review by gantunesr — APPROVED** [review]

**Pull request merged**